### DISCUSSION TITLE
Does ace.js have the ability to auto-detect languages?

### DISCUSSION THREAD

**Discussion by justanotherbyte** [ajaxorg/ace#4826] [Q&A] [answered]
Just wondering if ace.js had the ability to auto-detect a language based one some text we give it?

**Comment by andrewnester** [comment, marked as answer]
No, not really, the only option is to detect based on file path
https://stackoverflow.com/questions/15903191/how-to-automatically-pick-a-mode-for-ace-editor-given-a-file-extension
